### ISSUE TITLE
Checkout from remote opens branch in detached head

### ISSUE THREAD

**Issue by LanderMoerkerke** [jesseduffield/lazygit#833]
**Describe the bug**

If I checkout a remote branch (by pressing space on the branch itself) it opens the branch in a detached head state. 

If I want to merge a remote branch into my own branch, I'll have to manually checkout the branch outside lazygit.

I don't know if this behaviour is normal, my goal is to merge so what's the best/easiest way to do this?

**To Reproduce**

See screenshots.

**Expected behavior**

I want to merge a remote branch so it would be nice to checkout the remote branch, switch to my branch and merge it. Detached head branches will disappear when you switch to another branch.

**Screenshots**

Starting point:
![image](https://user-images.githubusercontent.com/20244592/81587639-949f7d80-93b7-11ea-86ae-30dec34cea55.png)

Switch to remotes:
![image](https://user-images.githubusercontent.com/20244592/81587671-a1bc6c80-93b7-11ea-9678-bc1f463c590b.png)

Press space to checkout:
![image](https://user-images.githubusercontent.com/20244592/81587728-bc8ee100-93b7-11ea-867b-a67c37c9d7f8.png)

Result:
![image](https://user-images.githubusercontent.com/20244592/81587748-c31d5880-93b7-11ea-9651-8ac7b33771bf.png)

**Desktop (please complete the following information):**
 - OS: Linux
 - Lazygit Version v0.20.2 (also had this on other versions)
 - The last commit id if you built project from sources: 922d8662ff0962058abd2f66c86f0e79bbba0ba1
- git version 2.26.2

Thanks!

**Comment by lordlycastle** [comment]
I wish I knew Go to add this to the project. Come on guys, let's pair up and use all the minds! #RemoteFTW

**Comment by markonyango** [comment]
You can always hit 'n' to create a new branch locally, that has the same name and make git choose the selected remote branch as the upstream. That does exactly what you want. AFAIK all git tools will do the following when checking out remote branches:

1. Create a new branch locally (if it doesn't already exist)
2. Set the remote branch of interest as the upstream for the local branch
3. Checkout the new local branch

Please correct me if I am wrong.

**Comment by lordlycastle** [comment]
Nope. It doesn’t create new branch and set its upstream. It just checks it out and point detached head to the Latest commit.  

We want a single operation to checkout new branches from remotes. Not 3 steps.

**Comment by Rizhiy** [comment]
@markonyango Hi, I tried your method and 'n' doesn't use the same name, it prepends the name of the remote to the branch name. As a result you can't easily push the branch back to remote since the name doesn't match.

As others have said 'checkout' should mirror what plain 'git' does, i.e. create branch with exactly the same name locally, make it track the remote and pull the latest changes.
